src/List.tsx

```diff
@@ -340,17 +346,41 @@ class List<Value = string> extends React.Component<IProps<Value>> {
     });
   };
 
-  onEnd = (e: Event) => {
+  isDraggedItemOutOfBounds = () => {
+    const initialRect = this.getChildren()[
+      this.state.itemDragged
+    ].getBoundingClientRect();
+    const targetRect = this.ghostRef.current!.getBoundingClientRect();
+    if (Math.abs(initialRect.left - targetRect.left) > targetRect.width) {
+      if (this.state.itemDraggedOutOfBounds === -1) {
+        this.setState({ itemDraggedOutOfBounds: this.state.itemDragged });
+      }
+      return true;
+    }
+    if (this.state.itemDraggedOutOfBounds > -1) {
```

Comment:
I was a bit unsure why state is reset here. calling `isDraggedItemOutOfBounds` a second time is required to get back to default? My first instinct would be to reset after this value is consumed. Interested to learn why this was positioned here

Reply:
`isDraggedItemOutOfBounds` is constantly called when an item is being moved around and then one more time when dropped. In both cases, `this.state.sDraggedItemOutOfBounds` can be reseted so that's why I put it directly into this method so it can be done only in one place.

Reply:
When an item is being moved around, it can be be flipped to `> -1` and back multiple times.

Reply:
thanks!